src/v8/v8.cc

```diff
@@ -627,15 +631,72 @@ void V8::getModuleFunctionImpl(std::string_view function_name,
     SaveRestoreContext saved_context(context);
     auto trap = func->call(params, results);
     if (trap) {
-      fail(FailState::RuntimeError, "Function: " + std::string(function_name) + " failed: " +
-                                        std::string(trap->message().get(), trap->message().size()));
+      auto message = this->get_fail_message(std::string(function_name), std::move(trap));
+      fail(FailState::RuntimeError, message);
       return R{};
     }
     R rvalue = results[0].get<typename ConvertWordTypeToUint32<R>::type>();
     return rvalue;
   };
 }
 
+std::string V8::get_fail_message(std::string function_name, wasm::own<wasm::Trap> trap) {
```

Comment:
I think `function_name` could be passed as `string_view` in the arguments, and just converted inside the function body.